ci(benchmarks): create new AST for each benchmark run

overlookmotel opened the pull request:
Make 2 changes to `oxc_semantic` benchmark:

1. Re-use a single allocator.
2. Create a fresh AST (in that same allocator) for each benchmark run - because `SemanticBuilder` mutates the AST.

Hopefully this will reduce noise in the benchmark a bit, and is a closer analog to real-world usage.

overlookmotel commented:
No significant difference in bench results. Good. Hopefully it will reduce variance a bit, though - we'll have have to see that over time.

Boshen commented:
### Merge activity

* **Feb 26, 8:08 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
ci(benchmarks): create new AST for each benchmark run (#9391)

Make 2 changes to `oxc_semantic` benchmark:

1. Re-use a single allocator.
2. Create a fresh AST (in that same allocator) for each benchmark run - because `SemanticBuilder` mutates the AST.

Hopefully this will reduce noise in the benchmark a bit, and is a closer analog to real-world usage.
